src/Common/test/Common.TestResources/CancellationTokenSourceExtensions.cs

```diff
@@ -11,7 +11,9 @@ public static class CancellationTokenSourceExtensions
     // Workaround for Sonar S6966 bug: CancelAsync is suggested, which isn't available in .NET 6.
     public static Task CancelAsync(this CancellationTokenSource source)
     {
+#pragma warning disable S3900 // Arguments of public methods should be validated against null
         source.Cancel();
+#pragma warning restore S3900 // Arguments of public methods should be validated against null
```

Comment:
```suggestion
        ArgumentNullException.ThrowIfNull(source);
        
        source.Cancel();
```
In case of null, we'd want the right error, not a NRE.